client/rpmtrans.c

```diff
@@ -211,13 +211,115 @@ TDNFPopulateTransaction(
     goto cleanup;
 }
 
+uint32_t
+TDNFDetectPreTransFailure(
+    rpmts pTS,
+    char *pszError
+    )
+{
+    uint32_t dwError = 0;
+    rpmtsi pi = NULL;
+    rpmte pte = NULL;
+    char *pszToken;
+    int i = 0;
+    char *pszPkgName = NULL;
+    char *pszSymbol = NULL;
+    char *pszVersion = NULL;
+    char *pszEVR = NULL;
+    char *pszCachePkgName = NULL;
+    char *pszCachePkgEVR = NULL;
+
+    if (!pTS || !pszError)
+    {
+        dwError = ERROR_TDNF_INVALID_PARAMETER;
+        BAIL_ON_TDNF_ERROR(dwError);
+    }
+    // Error Str has the format: <Pkg-Name> <Symbol> <version-release>
+    pszToken = strtok(pszError, " ");
```

Comment:
Please use strsep instead of strtok.
I think it's better if you do a strdup of pszError string before parsing it using strtok (strtok alters the input string)
So, if pszError is read only, we might end up in segfault.

Reply:
Since pszError is no longer used in this function as well as in the caller, I feel it should okay if strtok modifies it. what advantage do you see using strsep instead of strtok?

Reply:
Advantage is none in our case but strsep is better than strtok while handling empty fields: https://man7.org/linux/man-pages/man3/strsep.3.html

And concern about pszError is not just modifying, this function might grow in future and we may need the original error string and if in any library updates if error string becomes read only, tdnf will crash

Reply:
Done. Thanks